### PR TITLE
add support for directly executing without file

### PR DESCRIPTION
**Pull request by sudodaksh**
Added Support to programmatically executing code without requiring a file.

```python
from modiscript.api import ModiScript



ms = ModiScript(debug=False)

ms.execute("""
mitrooon

bhaiyo aur behno "Do whatever you want here"

acche din aa gaye
""", valueType="code")


```

### CODE REVIEW

File: modiscript/api.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+from .lexer import Lexer
+from .parser import Parser
+
+
+class ModiScript:
+    def __init__(self, debug=False):
+        self.debug = debug
+
+    def _compile_file(self, value, valueType="filename"):
```
Comment (onlinejudge95) [inline modiscript/api.py]: A small change but what do you think about using snake_case for `valueType`. Because Modiji **MIGHT** hate camel.

**Review by onlinejudge95 — COMMENTED** [review]
PR is functional, just some variable casing issue. They are lightweight but should preferably be there going that the compiler is in `python`

**Comment by sudodaksh** [comment]
Sure, I've fixed the casing to what modiji prefers ;)

**Comment by sudodaksh** [comment]
@onlinejudge95 The flask api is ready, I've made stdin and stdout work. but it depends on this pr, could you please process it?

**Comment by VPanjeta** [comment]
Hey @DakshMiglani 
Is there any special reason why you have taken the runnable file out of the directory?
The general structure of a project in compilers and interpreters is where the root folder does not have any binaries. they are generally stored in `bin` folder or a directory with the same as the project.
If you wanna take `MODI` file out of the lexer-parser kit then it is preferable that a runnable file is present in the `bin` folder and not in root.

**Comment by sudodaksh** [comment]
@Vpanjeta ok, I'll move it to the bin directory.

**Comment by sudodaksh** [comment]
@VPanjeta done.

**Comment by VPanjeta** [comment]
Thanks for the contribution.

**Comment by VPanjeta** [comment]
Actually just noticed that the binary file if taken out of the `modiscript` folder will only be able to run if `modiscript` is a python package. In your case it was a relative import so if you run MODI from bin folder it works but if you run it in any other directory that does not have `modiscript` directory in the relative path it won't work. It'll throw a package not found error since it is not installed through an egg file or pip and is not present in the python lib folder. 
Now there are 2 ways of doing this, either we let the MODI file be in the modiscript folder or we create a setup.py file for it that makes the package accessible to the MODI file by installing the package.
Since I do not have enough time to focus on it, I'd rather go for the former approach for now. Will probably make a package out of it when I have time. For the time being, I'm reverting the commit in the master branch.

**Comment by sudodaksh** [comment]
@VPanjeta  Can't we instead make a install script, that just installs the python module from the local directory, which can then be used in the binary file? Let me know, I can submit the pr.

**Pull request merged**